internal/service/flexcluster/data_source_schema.go

```diff
@@ -0,0 +1,108 @@
+package flexcluster
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func DataSourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Attributes: dataSourceSchema(false),
+	}
+}
+
+func dataSourceSchema(isPlural bool) map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"project_id": schema.StringAttribute{
+			Required:            !isPlural,
+			Computed:            isPlural,
```

Comment:
project_id will be required in both plural and singular data source. [List operation endpoint](https://www.mongodb.com/docs/openapi/preview/?src=https://raw.githubusercontent.com/mongodb/atlas-sdk-go/dev-latest/openapi/atlas-api.yaml#tag/Flex-Clusters/operation/listFlexClusters) needs project_id

Reply:
Good catch, fixed!

Reply:
@oarbusi is it possible the first definition is correct? Given that this project_id is the one defined at each flex cluster element, for the case of the plural data source it would be computed and the required project_id is defined at the root level of the schema.

Reply:
Very good point, didn't think about this not being at the root level on the plural data source

Reply:
Noted, this change has been reverted!